StRoot/StSpinPool/StFcsTriggerSimMaker/StFcsTriggerSimMaker.cxx

```diff
@@ -279,43 +305,83 @@ int StFcsTriggerSimMaker::Make(){
     memset(fcs_trg_sim_adc,0,sizeof(fcs_trg_sim_adc));
     int n=0;
     for(int det=0; det<=kFcsNDet; det++){
-	StSPtrVecFcsHit& hits = mFcsColl->hits(det);
-	int ns  = mFcsDb->northSouth(det);
-	int ehp = mFcsDb->ecalHcalPres(det);
-	int nh  = mFcsColl->numberOfHits(det);
-	for(int i=0; i<nh; i++){
-	    StFcsHit* hit=hits[i];
-	    unsigned short dep = hit->dep();
-	    unsigned short ch  = hit->channel();
-	    //printf("ns=%1d ehp=%1d dep=%2d ch=%2d adc=%4d sim=\n",ns,ehp,dep,ch,hit->adc(0),mSimMode);
-	    if(ehp<0 || ch>=32) continue;
-	    fcs_trg_sim_adc[ns][ehp][dep][ch] = hit->adc(0);
-	    if(mSimMode==0){
-	      int ntb=hit->nTimeBin();
-	      for(int t=0; t<ntb; t++){
-		int tb = hit->timebin(t);
-		if(tb>=mTrgTimebin-3 && tb<=mTrgTimebin+4){
-		  data[tb-mTrgTimebin+3] = hit->adc(t);
-		  //printf("tb=%3d i=%2d adc=%4d\n",tb,tb-mTrgTimebin+3,hit->adc(t));
-		}
-	      }
-	      mTrgSim->fill_event(ehp,ns,dep,ch,data,8) ;
-	    }else{
-	      data[1] = hit->adc(0)-1;  //removing 1 to add at tb6
-	      data[6] = 1;              //add this so tb6>tb7
-	      mTrgSim->fill_event(ehp,ns,dep,ch,data,8) ;
-	    }
-	    if(mFile) fprintf(mFile,"%2d %2d %2d %2d %5d\n",ns,ehp,dep,ch,hit->adc(0));
-	    n++;
-	}
+        
+        int ns  = mFcsDb->northSouth(det);
+        int ehp = mFcsDb->ecalHcalPres(det);
+                
+        if((event)&&(mFcsColl)){
+            StSPtrVecFcsHit& hits = mFcsColl->hits(det);
+        
+            int nh  = mFcsColl->numberOfHits(det);
+          
+            for(int i=0; i<nh; i++){
+                StFcsHit* hit=hits[i];
+                unsigned short dep = hit->dep();
+                unsigned short ch  = hit->channel();
+//                printf("ns=%1d ehp=%1d dep=%2d ch=%2d adc=%4d sim=%d\n",ns,ehp,dep,ch,hit->adc(0),mSimMode);
+                if(ehp<0 || ch>=32) continue;
+                fcs_trg_sim_adc[ns][ehp][dep][ch] = hit->adc(0);
+                if(mSimMode==0){
+                  int ntb=hit->nTimeBin();
+                  for(int t=0; t<ntb; t++){
+                int tb = hit->timebin(t);
+                if(tb>=mTrgTimebin-3 && tb<=mTrgTimebin+4){
+                  data[tb-mTrgTimebin+3] = hit->adc(t);
+//                  printf("tb=%3d i=%2d adc=%4d\n",tb,tb-mTrgTimebin+3,hit->adc(t));
+                }
+                  }
+                  mTrgSim->fill_event(ehp,ns,dep,ch,data,8) ;
+                }else{
+                  data[1] = hit->adc(0)-1;  //removing 1 to add at tb6
+                  data[6] = 1;              //add this so tb6>tb7
+                  mTrgSim->fill_event(ehp,ns,dep,ch,data,8) ;
+                }
+                if(mFile) fprintf(mFile,"%2d %2d %2d %2d %5d\n",ns,ehp,dep,ch,hit->adc(0));
+                n++;
```

Comment:
Would you mind moving the common code into a separate function?

Reply:
Sure thing. I will go ahead and make the changes thank you for the suggestion.

Reply:
I just added my modification after your suggestions to my local fork of the star-sw page. If i go to the initial pull request that I made I can see the commit with the new changes there but I'm not sure if those would be come the new ones here. Can you please tell me if you see the new commits and whether the updates follow your suggestions.

Reply:
I will mark this one as resolved.